workflow/automation/lib/schedulers/pbs.py

```diff
@@ -173,6 +174,30 @@ def check_queues(self, user: bool = False, target_machine=None) -> List[str]:
         self.logger.debug(output_list)
         return output_list
 
+    def check_wct(self, job_id: int):
+        """
+        Checks the given job_id if it has failed due to Wall Clock Time
+        :param job_id: The id of the job to be checked for wct
+        :return: Boolean for if the job has failed due to Wall Clock Time or not
+        """
+        cmd = f"qstat -x {job_id} -f -F json | grep walltime"
+        output, err = self._run_command_and_wait(cmd=[cmd], shell=True)
+
+        walltime_split = output.split()
+        _, limit_hour, limit_min, limit_sec = (
+            walltime_split[1].replace('"', "").split(":")
+        )
+        limit_time = timedelta(
+            hours=int(limit_hour), minutes=int(limit_min), seconds=int(limit_sec)
+        )
+        _, elapsed_hour, elapsed_min, elapsed_sec = (
+            walltime_split[0].replace('"', "").split(":")
+        )
+        elapsed_time = timedelta(
+            hours=int(elapsed_hour), minutes=int(elapsed_min), seconds=int(elapsed_sec)
+        )
+        return elapsed_time > limit_time
```

Comment:
If it's equal it also hit WCT. In the case it finished exactly on time, I think most jobs have a short circuit to mark them as done once they start.